pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/Server2ControllerSegmentUploader.java

```diff
@@ -24,6 +24,7 @@
 import org.apache.pinot.common.metrics.ServerMetrics;
 import org.apache.pinot.common.protocols.SegmentCompletionProtocol;
 import org.apache.pinot.common.utils.FileUploadDownloadClient;
+import org.apache.pinot.common.utils.LLCSegmentName;
```

Comment:
In this class, we should not need the segment name argument anymore since we get it in the upload() call.

Reply:
This is a compile time dependency because this uploader implements the interface's 
    URI uploadSegment(File segmentFile, LLCSegmentName segmentName);
method.  The interface method uses segmentName -- which is a reasonable choice given that some implementation may need that info e.g., PinotFSUploader.

Reply:
I meant that we shouldnot need the segment name argument in the constructor of the class. Sorry, I should have been clearer. You can do this in another PR. It is not a public interface, so we are ok